subprojects/language-java/src/main/java/org/gradle/external/javadoc/CoreJavadocOptions.java

```diff
@@ -701,4 +704,16 @@ public final void write(File outputFile) throws IOException {
     public JavadocOptionFileOption<File> addFileOption(String option, File value) {
         return optionFile.addFileOption(option, value);
     }
+
+    /**
+     * This method exists so that changing any options to the Javadoc task causes it to be re-run.
+     *
+     * @return the complete list of options, converted to strings
+     * @since 7.5
+     */
+    @Incubating
+    @Input
+    protected Map<String, String> getFullOptions() {
```

Comment:
This works but at the cost of cache hits.  There are options here that are file paths that are considered as inputs elsewhere (e.g., `stylesheetFile`). 

We didn't have very much coverage of this, so I added some in https://github.com/gradle/gradle/pull/20041

Is there some way we could only capture the "extra" options vs all of the options that may already be inputs?

Reply:
@big-guy - I filtered the new input property to only include the unknown extra properties.  I think this should enable previously cached data to be found with this change (if no extra options are provided).  All the new tests you added are merged into this and pass.